modules/middlewares/redis.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2013 Beego Authors
+// Copyright 2014 The Macaron Authors
+// Copyright 2020 The Gitea Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"): you may
+// not use this file except in compliance with the License. You may obtain
```

Comment:
looks like a copy of https://gitea.com/go-chi/session/src/branch/master/redis/redis.go

do we have to have it twice?

Reply:
Copied from modules/session/redis.go and should remove that one until macaron removed. I will add comment here.